### DISCUSSION TITLE
Will MOJO have Unions, lile C or C++?

### DISCUSSION THREAD

**Discussion by joaocarvalhoopen** [modular/modular#773] [General]
Hello,

To map types from C and C++ to MOJO, MOJO will have to have C style Unions, compatible with the C ABI,
So my question is will it have, or when will it have Union's?
Is this a priority for the MOJO implementation?

Best regards,

**Comment by Mogball** [comment]
Hey, although C-style unions are not on the roadmap themselves, ADTs are. Implementing language support for C-style unions is one path to implementing ADTs in the language, and it's not a bad to have support for C-style unions anyways. I'm not able to provide a timeline at the moment, however.

**Reply by joaocarvalhoopen** [reply]
Thank you very much Mogball.

Best regards,
